Assets/Scripts/PlayerCollision.cs

```diff
@@ -26,53 +50,53 @@ private void OnCollisionEnter2D(Collision2D col)
 		{
 			QuantumGyroBlade selfQGB = GetComponent<MovementController>().QGB;
 			QuantumGyroBlade opponentQGB = other.GetComponent<MovementController>().QGB;
-			Feedback(other, col.contacts[0].point, _rigidbody2D.velocity.magnitude + rb.velocity.magnitude);
-			float mag = _rigidbody2D.velocity.magnitude;
-			_rigidbody2D.velocity = (transform.position - other.transform.position)
-			                        * rb.velocity.magnitude
-			                        * CollisionDamping
-			                        * opponentQGB.Power
-			                        / selfQGB.Resistance;
-			rb.velocity = (other.transform.position - transform.position)
-			              * mag
-			              * CollisionDamping
-			              * selfQGB.Power
-			              / opponentQGB.Resistance;
+			Feedback(other, _rigidbody2D.velocity.magnitude + rb.velocity.magnitude);
+			float a1 = Vector2.Angle(rb.velocity, transform.position - other.transform.position);
+			float a2 = Vector2.Angle(_rigidbody2D.velocity, transform.position - other.transform.position);
+			if(a1 > a2)
```

Comment:
So after reading what `DoCollision` does it looks like there isn't any difference between who is `instigator` and who is `receiver`. Am I correct in that respect?
If I am correct, what is the purpose of doing this unnecessary computation?

Reply:
This is not currently being used, but will be used in the future for networking collisions as only one of the players should do the collision calculation. It also opens design space for abilities relying one being one or the other.